swagger.yaml

```diff
@@ -1,75 +1,164 @@
 swagger: '2.0'
+
 info:
-  title: Web API Skeleton
-  description: Skeleton for Dropwizard Web APIs
-  version: '0.0'
-  license:
-    name: GNU Affero General Public License Version 3
-    url: http://www.gnu.org/licenses/agpl-3.0.en.html
-externalDocs:
-  description: GitHub Repository
-  url: https://github.com/osu-mist/web-api-skeleton
-host: localhost
+  description: "This API retrieves CORE permissions information on different users."
+  version: "1.0"
+  title: "CORE Permissions API"
+
+tags:
+- name: "permissions"
+  description: "HR, Student, and Finance permissions (1-5)"
+
 schemes:
-  - http
-basePath: /api/v0
+- "https"
+
+host: "api.oregonstate.edu"
+
+basePath: "/v1"
+
+externalDocs:
+  description: "GitHub Repository"
+  url: "https://github.com/osu-mist/core-permissions-api"
+
 consumes:
-  - text/plain
+  - "application/json"
+
 produces:
-  - text/plain
+  - "application/json"
+
 paths:
-  /:
+
+  /permissions/{osuID}:
     get:
-      summary: Get information
-      description: Build and runtime information about the application.
+      tags:
+      - "permissions"
+      summary: "Retrieve permissions for user by their OSU ID"
+      operationId: "getPermissionsById"
       parameters:
-        - $ref: '#/parameters/pretty'
+        - name: "osuID"
+          in: "path"
+          description: "OSU ID of person whose permissions will be retrieved"
+          required: true
+          type: "string"
+
       responses:
         200:
-          description: Object containing information
+          description: "Permissions successfully retrieved"
           schema:
-            $ref: '#/definitions/Info'
+            $ref: "#/definitions/PermissionsResult"
+        404:
+          description: "No permissions found for this OSU ID"
+          schema:
+            $ref: "#/definitions/Error"
+        500:
+          description: "Internal server error"
+          schema:
+            $ref: "#/definitions/Error"
 
-parameters:
-  pretty:
-    name: pretty
-    in: query
-    type: boolean
-    required: false
-    description: If true, JSON response will be pretty-printed
+  /permissions:
+    get:
+      tags:
+      - "permissions"
+      summary: "Retrieve permissions for users by parameters"
+      operationId: "getPermissions"
+      parameters:
+        - name: "osuID"
+          in: "query"
+          description: "OSU ID of person whose permissions will be retrieved"
+          required: false
+          type: "string"
+        - name: "onid"
+          in: "query"
+          description: "ONID of person whose permissions will be retrieved"
+          required: false
+          type: "string"
+      responses:
+        200:
```

Comment:
What if no person is found? Will the response be like `"data": []` or `"data": null`? Either way [could work within JSON API spec](http://jsonapi.org/format/#fetching-resources), just curious how you are designing it.

Reply:
I was thinking that the response would be like `"data": []` if no person is found.